sdk/storage/MigrationGuide.md

```diff
@@ -0,0 +1,477 @@
+```
+
+#### Connection string
+
+The following code assumes you have acquired your connection string (you can do so from the Access Keys tab under Settings in your Portal Storage Account blade). It is recommended to store it in an environment variable. Below demonstrates how to parse the connection string in v7.5 vs v12.
+
+v7.5
+
+```C++
+cloud_storage_account storage_account = cloud_storage_account::parse(storage_connection_string);
+cloud_blob_client service_client = storage_account.create_cloud_blob_client();
+```
+
+v12
+```C++
+BlobServiceClient serviceClient = BlobServiceClient::CreateFromConnectionString(connectionString);
+```
+
+You can also directly get a blob client with your connection string, instead of going through a service and container client to get to your desired blob. You just need to provide the container and blob names alongside the connection string.
+
+```C++
+BlobClient blobClient = BlobClient::CreateFromConnectionString(connectionString, containerName, blobName);
+```
+
+#### Shared Key
+
+Shared key authentication requires the URI to the storage endpoint, the storage account name, and the shared key as a base64 string. The following code assumes you have acquired your shared key (you can do so from the Access Keys tab under Settings in your Portal Storage Account blade). It is recommended to store it in an environment variable.
+
+Note that the URI to your storage account can generally be derived from the account name (though some exceptions exist), and so you can track only the account name and key. These examples will assume that is the case, though you can substitute your specific account URI if you do not follow this pattern.
+
+v7.5
+```C++
+cloud_blob_client blob_client(storage_uri(blob_url), storage_credentials(account_name, account_key));
+```
+
+v12
+```C++
+auto credential = std::make_shared<StorageSharedKeyCredential>(accountName, accountKey);
+BlobServiceClient serviceClient(blobServiceUrl, credential);
+```
+
+If you wish to rotate the key within your `BlobServiceClient` (and any derived clients), you must retain a reference to the `StorageSharedKeyCredential`, which has the instance method `Update(std::string accountKey)`.
+
+### Client Structure
+
+**The legacy SDK used a stateful model.** There were container and blob objects that held state regarding service resources and required the user to manually call their update methods. But blob contents were not a part of this state and had to be uploaded/downloaded whenever they were to be interacted with. This became increasingly confusing over time, and increasingly susceptible to thread safety issues.
+
+The modern SDK has taken a client-based approach. There are no objects designed to be representations of storage resources, but instead clients that act as your mechanism to interact with your storage resources in the cloud. **Clients hold no state of your resources.** This is most noticable when looking at [blob metadata](#managing-blob-metadata).
```

Comment:
Please specify that `LeaseClient`s are exceptions.

Reply:
I agree lease state/status/period is **state** of a blob. But lease ID, it's more like a resource, maybe?

Reply:
`**Clients hold no state of your resources.**` but `LeaseClient` holds the state of your lease, which should be mentioned as an exception.

Reply:
all right, fine.